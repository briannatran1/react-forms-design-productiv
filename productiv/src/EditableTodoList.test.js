import React from "react";
import { render } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";

describe('EditableTodoList', function () {
  const todos = [{
    id: 1,
    title: 'testing',
    description: 'finish testing',
    priority: 2
  },
  {
    id: 2,
    title: 'eat',
    description: 'eat lunch',
    priority: 1
  }
  ];

  test('renders without crashing', function () {
    render(<EditableTodoList todos={todos} />);
  });

  test('matches snapshot', function () {
    const { asFragment } = render(
      <EditableTodoList todos={todos} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('shows list of todos', function () {
    const { queryByText, queryAllByText } = render(<EditableTodoList todos={todos} />);

    expect(queryByText('testing')).toBeInTheDocument();
    expect(queryByText('(Priority: 2)')).toBeInTheDocument();
    expect(queryByText('finish testing')).toBeInTheDocument();

    expect(queryByText('eat')).toBeInTheDocument();
    expect(queryByText('(Priority: 1)')).toBeInTheDocument();
    expect(queryByText('eat lunch')).toBeInTheDocument();

    expect(queryAllByText('Edit').length).toEqual(2);
    expect(queryAllByText('Del').length).toEqual(2);
  });
});