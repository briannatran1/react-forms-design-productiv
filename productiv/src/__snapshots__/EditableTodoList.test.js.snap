// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTodoList matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              testing
            </b>
             
            <small>
              (Priority: 2)
            </small>
          </div>
          <div>
            <small>
              finish testing
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              eat
            </b>
             
            <small>
              (Priority: 1)
            </small>
          </div>
          <div>
            <small>
              eat lunch
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
